add finnish paraphrase-based STS dataset

rbroc opened the pull request:
## Checklist for adding MMTEB dataset

Reason for dataset addition: First Finnish STS dataset

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

rbroc commented:
A note on this: performance scores are much lower than other STS tasks. I have run an additional test though, testing a monolingual finnish model, https://huggingface.co/TurkuNLP/sbert-uncased-finnish-paraphrase, which gets similarity scores between .3 - .4. 

Furthermore, the similarity scores obtained with multilingual encoders should be statistically significant (p ~ 0.001), if that's a good definition of better than chance.

In my view, just a challenging dataset (most labels are 3 and 4, and the different between the two is `3 = good paraphrase in this context`, but not in general; `4 = good paraphrase in all contexts`), just calling for a very nuance semantic understanding.

But happy to hear what y'all think.

isaac-chung commented on mteb/tasks/STS/fin/FinParaSTS.py:
```diff
@@ -0,0 +1,75 @@
+            abstract = "In this paper, we introduce the first fully manually annotated paraphrase corpus for Finnish containing 53,572 paraphrase pairs harvested from alternative subtitles and news headings. Out of all paraphrase pairs in our corpus 98{\%} are manually classified to be paraphrases at least in their given context, if not in all contexts. Additionally, we establish a manual candida ...
+        }        
+        """,
+        n_samples={"test": 1000, "validation": 1000},
```
nit: use a variable `n_samples=1000`

rbroc replied:
pydantic throws an error, as this must be a dictionary. will keep it as is :)

isaac-chung replied:
Oh sorry. I meant something like this, and defining `N_SAMPLES=1000` around line 6 or 7.
```suggestion
        n_samples={"test": N_SAMPLES, "validation": N_SAMPLES},
```

isaac-chung commented on mteb/tasks/STS/fin/FinParaSTS.py:
```diff
@@ -0,0 +1,75 @@
+    def dataset_transform(self):
+        self.dataset = self.dataset.shuffle(seed=42)
+        for split in self.dataset:
+            self.dataset[split] = self.dataset[split].select(range(1000))
```
same here for 1000.

rbroc replied:
resolving as per previous comment

isaac-chung reviewed (approved):
Looks good to me, but I would like a second opinion from one of the maintainers: @KennethEnevoldsen or @imenelydiaker 
 maybe. 
I'm unsure (for my own curiosity) about the [extra label 4 flags](https://aclanthology.org/2021.nodalida-main.29.pdf)' role in the scoring:
```
- Subsumption (> or <) w
- Style (s)
- Minor deviation (i)
```

rbroc commented:
thanks a lot @isaac-chung! Re: the extra labels, I have simply stripped additional annotations, and left the raw scores as labels, but curious to hear what people think. :)

imenelydiaker commented on mteb/tasks/STS/fin/FinParaSTS.py:
```diff
@@ -0,0 +1,75 @@
+        rename_dict = {"text1": "sentence1", "text2": "sentence2", "label": "score"}
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.select_columns(list(rename_dict.values()))
+        self.dataset = self.dataset.map(lambda x: {"score": int(x["score"][0])})
```
Very nice! You're handling the string `label` conversion to integers 🙂

rbroc replied:
we're pulling main now, hence ready to resolve

rbroc commented on mteb/tasks/STS/fin/FinParaSTS.py:
```diff
@@ -0,0 +1,75 @@
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.shuffle(seed=42)
+        for split in self.dataset:
+            self.dataset[split] = self.dataset[split].select(range(1000))
+        rename_dict = {"text1": "sentence1", "text2": "sentence2", "label": "score"}
+        self.dataset = self.dataset.rename_columns(rename_dict)
+        self.dataset = self.dataset.select_columns(list(rename_dict.values()))
+        self.dataset = self.dataset.map(lambda x: {"score": int(x["score"][0])})
```
@imenelydiaker self.dataset is a `DatasetDict`, and `select` cannot be applied to that -- we could do this as part of map, but it would be cumbersome and subsetting early on increases efficiency. 
would strongly lean towards leaving as is.

rbroc commented:
> Is this dataset a subset/duplication of [GEM/opusparcus](https://huggingface.co/datasets/GEM/opusparcus)? In which case, Opusparcus is already handled in mteb and handles Finnish. Can you please check? Because I see a `gem_id`, which refers to the id of the item in the GEM dataset, maybe we can filter samples that are already handled by Opusparcus using the `gem_id` column ?

I have double-checked and it is not. `GEM` refers to the data format for OpenSubtitles, which both use to get `sentence1` (though very different subsets). Paraphrases are then generate with automated translation of parallel examples for opusparcus, while here they use alternative versions of Finnish subtitles. Also, this dataset includes news on top of subtitles.

rbroc commented on mteb/tasks/STS/fin/FinParaSTS.py:
```diff
@@ -0,0 +1,75 @@
+        return metadata_dict
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.shuffle(seed=42)
```
loading `plain` now (i was loading `classification` before). it is done by passing `name` to the `dataset` dict in the metadata, as `dataset` is passed as kwargs to `load_data` (see https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/AbsTask.py#L35), and `name` specifies the config to pull. this should be ready to be resolved.

rbroc commented:
@imenelydiaker @isaac-chung I've addressed your points and resolved some of the things that can't be addressed (e.g., can't pass `n_samples=1000`, `pydantic` requires a dict). I've rerun evaluations too and added points.

isaac-chung commented:
Thanks! I clarified on my comments, but this still looks good to me. If it's all good from @imenelydiaker's side too then we can merge :)

imenelydiaker reviewed (approved):
Thanks for the clear answers, great job! We can merge 🙂

rbroc commented:
thanks @imenelydiaker and @isaac-chung for a very thorough review 🚀 I have made one more tiny change, passing `n_samples` as a variable as suggested by @isaac-chung (sorry for the misunderstanding! :)) and updating avg length after having decided to pull the `plain` subset. 

should be ready to merge as far as I am concerned too :)

rbroc commented:
@imenelydiaker @isaac-chung now tests on 3.8 are failing, but the errors do not seem to be dependent on anything related to this PR. it seems like it's skipping one the workflows though as a consequence of this one failing.

looks like it's something funky with caching though, can one of your trigger the workflows again to see if it's just an accident?

imenelydiaker commented:
I think it's related to this issue: https://github.com/embeddings-benchmark/mteb/issues/454. I see in the logs that it's related to markdown, maybe we can bypass it?

cc @KennethEnevoldsen

imenelydiaker commented:
I restarted the windows job with debug meanwhile

rbroc commented:
awesome! everything's passing now :)

imenelydiaker commented:
I'm merging, thank you again for your contribution @rbroc !
